src/Agent.Listener/SelfUpdater.cs

```diff
@@ -258,7 +315,12 @@ private async Task DownloadLatestAgent(CancellationToken token)
 
                 if (!downloadSucceeded)
                 {
-                    throw new TaskCanceledException($"Agent package '{archiveFile}' failed after {Constants.AgentDownloadRetryMaxAttempts} download attempts");
+                    string exceptionMessage = $"Agent package '{archiveFile}' failed after {Constants.AgentDownloadRetryMaxAttempts} download attempts.";
+                    if (!_hashValidationDisabled)
+                    {
+                        exceptionMessage += $"\nYou can try to set the environment variable DISABLE_HASH_VALIDATION=true to avoid this error, but this is not secure.";
```

Comment:
Let's also describe possible reasons when checksum validation failed - to give heads up on this:
 1) The agent package was compromised
 2) The agent package was not fully downloaded or was corrupted during download process
 
"You can skip checksum validation for agent package by ... "

Reply:
Done, now it looks like this:
```
Scanning for tool capabilities.
Connecting to the server.
2021-09-23 15:11:04Z: Listening for Jobs
Downgrading agent to a lower version. This is usually due to a rollback of the currently published agent for a bug fix. To disable this behavior, set environment variable AZP_AGENT_DOWNGRADE_DISABLED=true before launching your agent.
Agent update in progress, do not shutdown agent.
Downloading 2.192.0 agent
Error reported in diagnostic logs. Please examine the log for more details.
    - D:\agent\_cloud\win-x64\_diag\Agent_20210923-151059-utc.log
An error occurred: Agent package checksum validation failed.
There are possible reasons why this happened:
  1) The agent package was compromised.
  2) The agent package was not fully downloaded or was corrupted during the download process.
You can skip checksum validation for the agent package by setting the environment variable DISABLE_HASH_VALIDATION=true
```